### PR TITLE
add defopt

### PR DESCRIPTION
**Pull request by nh13**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by nh13** [comment]
@conda-forge-admin, please ping team

**Comment by nh13** [comment]
I happy to maintain this recipe

**Comment by nh13** [comment]
All checks are passing, ready for review.

File: recipes/defopt/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    - setuptools_scm
+  run:
+    - docutils >=0.10
+    - python >=3.5
+    - sphinxcontrib-napoleon >=0.7.0
+    - colorama
+
+test:
+  imports:
+    - _defopt_version
```
Comment (scopatz) [inline recipes/defopt/meta.yaml:30]: Is this part of the official API?

Comment (nh13) [reply recipes/defopt/meta.yaml:30]: I used `grayskull` as [recommended in the README](https://github.com/conda-forge/staged-recipes#grayskull---recipe-generator-for-python-packages-on-pypi).  I believe it's pulling this import from https://github.com/anntzer/defopt/blob/66731cf25496f515303587744fc9ce604463acde/setup.py#L16.  We can leave it out, as I dont' think it's strictly necessary.  Thoughts?

Comment (scopatz) [reply recipes/defopt/meta.yaml:30]: please remove

Comment (nh13) [inline recipes/defopt/meta.yaml:30]:
```suggestion
```

Comment (nh13) [reply recipes/defopt/meta.yaml]: Done

**Comment by nh13** [comment]
Closing and re-opening to trigger builds (hopefully)

**Comment by nh13** [comment]
Reason was [one of the checks if failing](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=199196&view=logs&j=c572e5d5-7ba7-55ce-5575-7ce1d9f7161e&t=fe245b22-2f08-5a76-43b2-e02a18e3fefa&l=111), but for a reason with conda forge CI setup:
```
+++ export _CE_M=
+++ _CE_M=
+++ export _CE_CONDA=
+++ _CE_CONDA=
+++ export CONDA_PYTHON_EXE=/usr/local/miniconda/bin/python
+++ CONDA_PYTHON_EXE=/usr/local/miniconda/bin/python
++ '[' 0 '!=' 0 ']'
++ __conda_hashr
++ '[' -n '' ']'
++ '[' -n '' ']'
++ hash -r
+ echo 'Configuring conda.'
+ setup_conda_rc ./ ./recipes ./.ci_support/osx64.yaml
Configuring conda.
Traceback (most recent call last):
  File "/usr/local/miniconda/bin/setup_conda_rc", line 7, in <module>
    from conda_forge_ci_setup.build_utils import setup_conda_rc
ModuleNotFoundError: No module named 'conda_forge_ci_setup'
```

**Comment by nh13** [comment]
Nope, same issue with the CI.  Google showed one other time here for this issue:
```
ModuleNotFoundError: No module named 'conda_forge_ci_setup'
```

**Comment by nh13** [comment]
@scopatz I think this is now ready for merge.  I think the build errors are not mine, so any help from the conda-forge/staged-recipes maintainers would be appreciated.

**Comment by nh13** [comment]
@conda-forge-admin, please ping team

**Comment by scopatz** [comment]
Thanks @nh13!

**Pull request merged**